Client/Extensions/PaymentRequestExtension.swift

```diff
@@ -0,0 +1,139 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import BraveShared
+import Data
+import Shared
+import WebKit
+
+private let log = Logger.browserLogger
+
+let popup = PaymentHandlerPopupView(imageView: nil, title: Strings.paymentRequestTitle, message: "")
+
+class PaymentRequestExtension: NSObject {
+    fileprivate weak var tab: Tab?
+    fileprivate var response = ""
+    fileprivate var token: String
+    
+    fileprivate enum PaymentRequestErrors: String {
+        case NotSupportedError = "NotSupportedError"
+        case AbortError = "AbortError"
+    }
+    
+    init(tab: Tab) {
```

Comment:
You should pass in `BraveRewards` here like other `TabContentScript`'s, instead of accessing through `Tab`